### PR TITLE
Improve assert error message in Process tests

### PR DESCRIPTION
**Pull request by stephentoub**
Contributes to https://github.com/dotnet/corefx/issues/35708

### CODE REVIEW

**Comment by danmosemsft** [comment]
Maybe we should have AssertExtensions forward everything on Assert. That way there's never any question which to choose, and we have the power to augment anything in Assert without modifying tests.

File: src/CoreFx.Private.TestUtilities/src/System/AssertExtensions.cs [hunk 1]
```diff
@@ -317,5 +318,14 @@ public static void Equal(byte[] expected, byte[] actual)
                 throw new AssertActualExpectedException(expectedString, actualString, null);
             }
         }
+
+        /// <summary>Validates that the two sets contains the same elements. XUnit doesn't display the full collections.</summary>
+        public static void Equal<T>(HashSet<T> expected, HashSet<T> actual)
```
Comment (ahsonkhan) [inline src/CoreFx.Private.TestUtilities/src/System/AssertExtensions.cs]: Don't we need to expose this from the ref?
https://github.com/dotnet/corefx/blob/908cba9d8f275935a9e91b32fd363b5687dc8f86/src/CoreFx.Private.TestUtilities/ref/CoreFx.Private.TestUtilities.cs#L15-L19

Comment (stephentoub) [reply src/CoreFx.Private.TestUtilities/src/System/AssertExtensions.cs]: Yes. Not sure how I managed to build locally. Must have built the wrong flavor.

Comment (danmosemsft) [reply src/CoreFx.Private.TestUtilities/src/System/AssertExtensions.cs]: Hah, that's probably why it doesn't build. I suspect neither of us tried building Linux.

**Pull request merged**